app/models/user.rb

```diff
@@ -116,10 +116,10 @@ def dummy_password_for_external_auth
   def change_password(oldpwd, newpwd)
     auth = self.class.authenticator(userid)
     raise MiqException::MiqEVMLoginError, "password change not allowed when authentication mode is #{auth.class.proper_name}" unless auth.uses_stored_password?
-    raise MiqException::MiqEVMLoginError, "old password does not match current password" unless User.authenticate(userid, oldpwd)
-
-    self.password = newpwd
-    self.save!
+    if User.authenticate(userid, oldpwd)
```

Comment:
What happens if the User *can't* authenticate at this point?  Is there an error about why given to the user?